integration/query_array_test.go

```diff
@@ -414,3 +414,185 @@ func TestArrayEquality(t *testing.T) {
 		})
 	}
 }
+
+// TestQueryArrayAll covers the case where the $all operator is used on an array or scalar.
+func TestQueryArrayAll(t *testing.T) {
+	t.Parallel()
+
+	ctx, collection := setup.Setup(t, shareddata.Composites, shareddata.Scalars)
+
+	// Insert additional data to check more complicated cases:
+	// - a longer array of ints;
+	// - a field is called differently and needs to be found with the {$all: [null]} case.
+	_, err := collection.InsertMany(ctx, []any{
+		bson.D{{"_id", "many-integers"}, {"customField", bson.A{42, 43, 45}}},
```

Comment:
Why not add it to shareddata?

Reply:
It doesn't seem to be needed in shared data.

Reply:
> It doesn't seem to be needed in shared data.

It should. 
IIRC we planned reordering and consistency for tests where all `InsertMany` is to be removed. Let's not add new work to that.

Reply:
@AlekSi as before we discussed that this doesn't go to shared data, please confirm or decline that @seeforschauer's statement.

Reply:
The current problem with shared data is that we have to update all tests if we add something to it. On the one hand, it is a feature, as it allows us to notice things that were incompatible with MongoDB before, but we did not notice them. On the other hand – yeah, we have to update a lot of tests.

The plan is to move all query tests to compatibility tests that will not have a list of IDs to update. That will give us benefit without the downside. For now, let's not update shared data.